docs/source/site-specific-config.rst

```diff
@@ -0,0 +1,142 @@
+.. _site_specific_config:
+
+Site-Specific Configuration
+***************************
+A site might have compilers that Fab doesn't know about, or prefers
+a different compiler from the Fab default. Fab abstracts the compilers
+and other programs required during building as an instance of a
+:class:`~fab.tools.tool.Tool` class. All tools that Fab knows about, are
+available in a :class:`~fab.tools.tool_repository.ToolRepository`.
+That will include tools that might not be available on the current system.
+
+Each tool belongs to a certain category of
+:class:`~fab.tool.categories.Categories`. A `ToolRepository` can store
+several instances of the same category.
+
+At build time, the user has to create an instance of
+:class:`~fab.tools.tool_box.ToolBox` and pass
+it to the :class:`~fab.build_config.BuildConfig` object. This toolbox
+contains all the tools that will be used during the build process, but
+it can only store one tool per category. If a certain tool should not
+be defined in the toolbox, the default from the `ToolRepository` will
+be used. This is useful for many standard tools like `git`, `rsync`
+etc that de-facto will never be changed.
+
+
+Categories
+==========
+All possible categories are defined in
+:class:`~fab.tool.categories.Categories`. If additional categories
+should be required, they can be added.
+
+Tool
+====
+Each tool must be derived from :class:`~fab.tools.tool.Tool`.
+The base class provides a `run` method, which any tool can
+use to execute a command in a shell. Typically, a tool will
```

Comment:
You can easily overwrite the run function of the `Tool` base class and just call a Python function. I actually considered doing this when I noticed the templaterator, but decided against it since I would assume that the CLI will stay identical, while the implementation might change(?)

Reply:
Agreed.